src/commons/controlBar/ControlBarShareButton.tsx

```diff
@@ -1,211 +1,143 @@
-          <Tooltip content="Copy link to clipboard">
-            <CopyToClipboard text={this.props.queryString!}>
-              <ControlButton icon={IconNames.DUPLICATE} onClick={this.selectShareInputText} />
-            </CopyToClipboard>
-          </Tooltip>
-        </div>
-      ) : (
-        <>
-          {!this.state.isSuccess || this.props.shortURL === 'ERROR' ? (
-            !this.state.isLoading || this.props.shortURL === 'ERROR' ? (
-              <div>
-                {Constants.urlShortenerBase}&nbsp;
-                <input
-                  placeholder={'custom string (optional)'}
-                  onChange={this.handleChange}
-                  style={{ width: 175 }}
-                />
-                <ControlButton
-                  label="Get Link"
-                  icon={IconNames.SHARE}
-                  // post request to backend, set keyword as return uuid
-                  onClick={() => this.fetchUUID(this.props.token)}
-                />
-              </div>
-            ) : (
-              <div>
-                <NonIdealState
-                  description="Generating Shareable Link..."
-                  icon={<Spinner size={SpinnerSize.SMALL} />}
-                />
-              </div>
-            )
-          ) : (
-            <div key={this.state.keyword}>
-              <input defaultValue={this.state.keyword} readOnly={true} ref={this.shareInputElem} />
-              <Tooltip content="Copy link to clipboard">
-                <CopyToClipboard text={this.state.keyword}>
-                  <ControlButton icon={IconNames.DUPLICATE} onClick={this.selectShareInputText} />
-                </CopyToClipboard>
-              </Tooltip>
-            </div>
-          )}
-        </>
-      );
-
-    return (
-      <Popover
-        popoverClassName="Popover-share"
-        inheritDarkTheme={false}
-        content={shareButtonPopoverContent}
-      >
-        <Tooltip content="Get shareable link" placement={Position.TOP}>
-          <ControlButton label="Share" icon={IconNames.SHARE} onClick={() => this.toggleButton()} />
-        </Tooltip>
-      </Popover>
-    );
-  }
-
-  public componentDidUpdate(prevProps: ControlBarShareButtonProps) {
-    if (this.props.shortURL !== prevProps.shortURL) {
-      this.setState({ keyword: '', isLoading: false });
-    }
-  }
+  const generateLinkSicp = () => {
+    const hash = playgroundConfiguration.encodeWith(new UrlParamsEncoderDelegate());
+    const shortenedUrl = `${Links.playground}#${hash}`;
+    setShortenedUrl(shortenedUrl);
+  };
 
-  private toggleButton() {
-    if (this.props.handleGenerateLz) {
-      this.props.handleGenerateLz();
-    }
+  const generateLink = props.isSicp
+    ? generateLinkSicp
+    : Constants.playgroundOnly
+    ? generateLinkPlaygroundOnly
+    : generateLinkBackend;
 
-    // reset state
-    this.setState({ keyword: '', isLoading: false, isSuccess: false });
-  }
+  useHotkeys([['ctrl+w', generateLink]], []);
```

Comment:
In Windows, this is equivalent to cmd+w on macOS, which will close the browser tab.